ako-operator/controllers/utils.go

```diff
@@ -251,17 +291,54 @@ func getEnvVars(ako akov1alpha1.AKOConfig, aviSecret v1.Secret) []v1.EnvVar {
 		envVars = append(envVars, envVar)
 	}
 
-	cacertRef, ok := aviSecret.Data["certificateAuthorityData"]
-	for k, v := range SecretEnvVars {
-		if k == "CTRL_CA_DATA" && (!ok || string(cacertRef) == "") {
-			continue
-		}
+	envVars = append(envVars, v1.EnvVar{
+		Name: "POD_NAME",
+		ValueFrom: &v1.EnvVarSource{
+			FieldRef: &v1.ObjectFieldSelector{
+				FieldPath:  "metadata.name",
+				APIVersion: "v1",
+			},
+		},
+	})
+
+	// add logfile and pvc
+	if ako.Spec.PersistentVolumeClaim != "" {
+		envVars = append(envVars, v1.EnvVar{
+			Name:  "USE_PVC",
+			Value: "true",
+		})
+	}
+
+	envVars = append(envVars, v1.EnvVar{
+		Name:  "LOG_FILE_PATH",
+		Value: ako.Spec.MountPath,
+	})
+
+	envVars = append(envVars, v1.EnvVar{
+		Name:  "LOG_FILE_NAME",
+		Value: ako.Spec.LogFile,
+	})
+	envVars = append(envVars, v1.EnvVar{
+		Name: "POD_NAMESPACE",
+		ValueFrom: &v1.EnvVarSource{
+			FieldRef: &v1.ObjectFieldSelector{
+				FieldPath:  "metadata.namespace",
+				APIVersion: "v1",
+			},
+		},
+	})
+	return envVars
+}
+
+func getEnvVarsForGateway(ako akov1alpha1.AKOConfig) []v1.EnvVar {
+	envVars := []v1.EnvVar{}
+	for k, v := range ConfigMapEnvVarsGateway {
 		envVar := v1.EnvVar{
 			Name: k,
 			ValueFrom: &v1.EnvVarSource{
-				SecretKeyRef: &v1.SecretKeySelector{
```

Comment:
have we taken out secret as secret shouldn't be present as part of ENV variable? Is AKOO responsible for creating secret objects from values.yaml? How secret will be generated?

Reply:
I have taken out the env variable that was created from the avi secret. The avi secret is not created by AKOO. It has to be created by the user before applying AKOConfig CRD object.